File: tourneyAPI/Models/Entities/Characters/DiddyKong.cs
namespace Entities;

using Enums;

// Defines the competitive profile metadata for this playable character.
public class DiddyKong : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public DiddyKong()
    {
        Id = CharacterId.DiddyKong;
        characterName = CharacterName.DIDDY_KONG;
        archetype = Archetype.NINJA;
        fallSpeed = FallSpeed.FAST_FALLERS;
        weightClass = WeightClass.LIGHTWEIGHT;
        tierPlacement = TierPlacement.S;
    }

}

File: tourneyClient/src/models/entities/Characters/DiddyKong.ts
import { CharacterId } from "@/models/Enums/CharacterId";
import { Character } from "../Character"
import { CharacterName } from "@/models/Enums/CharacterName";
import { Archetype } from "@/models/Enums/Archetype";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { WeightClass } from "@/models/Enums/WeightClass";
import { TierPlacement } from "@/models/Enums/TierPlacement";

export const DiddyKong: Character =
{
    id: CharacterId.DiddyKong,
    characterName: CharacterName.DIDDY_KONG,
    archetype: Archetype.NINJA,
    fallSpeed: FallSpeed.FAST_FALLERS,
    weightClass: WeightClass.LIGHTWEIGHT,
    tierPlacement: TierPlacement.S
}

export default DiddyKong; 